### PR TITLE
Add User Confirmation

### PR DESCRIPTION
**Pull request by nuwandavek**
- [x] Basic flow works
- [x] Handle interrupt properly
- [x] Show content in the confirmation
- [x] Show good toast
- [x] Migration for new state vars?

---

ToDo: 
Style the content by highlighting based on language (highlight js maybe?)

### CODE REVIEW

**Comment by nuwandavek** [comment]
@ppsreejith Can you check this out? Good to merge I think (after migration). I want to do the actual syntax highlighting as a separate step. But I want you to check and see if there are obvious errors here.

File: web/src/app/userConfirmation.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { dispatch } from "../state/dispatch"
+import { getState } from "../state/store"
+import { sleep } from "../helpers/utils"
+import { toggleUserConfirmation } from "../state/chat/reducer"
+
+export async function getUserConfirmation({content}: {content: string}) {
+  const thread = getState().chat.activeThread
+  const activeThread = getState().chat.threads[thread]
+  const messages = activeThread.messages
+  const msgIDX = messages.findLastIndex((message: any) => message.role === 'tool' && message.action.status === 'DOING');
```
Comment (ppsreejith) [inline web/src/app/userConfirmation.ts]: `msgIDX` seems unused?

**Review by ppsreejith — APPROVED** [review]
Lgtm!

**Comment by ppsreejith** [comment]
Tested and lgtm. Merging now.

**Pull request merged**